### PR TITLE
17.0 pos backport vat label fix hamo

### PR DESCRIPTION
**Pull request by hamo-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hamo-odoo** [comment]
@qle-odoo Back porting `VAT -> Tax` specific targeting 17.0 because the 17.1 pr had other issues to be fix too, and also added commit specific for `l10n_in_pos` where the tax info was missing on ticket

File: addons/l10n_in_pos/static/src/overrides/models/order.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ import { patch } from "@web/core/utils/patch";
 patch(Order.prototype, {
     export_for_printing() {
         const result = super.export_for_printing(...arguments);
+        if (this.pos.company.country.code === 'IN') {
+            result.tax_details.forEach((tax)=>{
+                tax.tax.letter = tax.tax.description.substring(0, 4)
+            })
+        }
```
Comment (hamo-odoo) [inline addons/l10n_in_pos/static/src/overrides/models/order.js:13]: Well I wanted to do it from `tax_group_id.name` but that would be take repetitive load of data where I have to load again account_tax_group_by_id but we can do it easily saas-17.1 this is `17.0` specific PR

**Comment by hamo-odoo** [comment]
> Look good to me, but you can squash the commit. And give a good description to your commit

Thanks! I have squashed both commits and improved the commit message

File: addons/l10n_in_pos/static/src/overrides/models/order.js [hunk 2]
```diff
@@ -6,6 +6,11 @@ import { patch } from "@web/core/utils/patch";
 patch(Order.prototype, {
     export_for_printing() {
         const result = super.export_for_printing(...arguments);
+        if (this.pos.company.country.code === 'IN') {
+            result.tax_details.forEach((tax)=>{
```
Comment (qle-odoo) [inline addons/l10n_in_pos/static/src/overrides/models/order.js:10]:
```suggestion
            result.tax_details.forEach((tax) => {
```

**Review by caburj — COMMENTED** [review]
Can you check other localizations that have overrides over the modified template?

**Comment by hamo-odoo** [comment]
@caburj `l10n_gcc` is using it, other than that found none where the tax headers are playing role ( I hope not to miss any other localization)
image

File: addons/l10n_in_pos/static/src/overrides/models/order.js (hunk 1 above)
Comment (caburj) [reply addons/l10n_in_pos/static/src/overrides/models/order.js:13]: May I know where in the source code that makes sure that the `description` contains the characters we need here?

Comment (hamo-odoo) [reply addons/l10n_in_pos/static/src/overrides/models/order.js:13]: https://github.com/odoo/odoo/blob/17.0/addons/l10n_in/data/template/account.tax-in.csv#L383

**Comment by hamo-odoo** [comment]
@caburj Idk though need to look at the loading of fields but I m getting an array of [id, name] of `account.tax.group` when append the fields of `account.tax` with `tax_group_id`

But does the job without loading the model `account.tax.group`
image

File: addons/l10n_in_pos/static/src/overrides/models/pos_store.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/** @odoo-module */
+
+import { PosStore } from "@point_of_sale/app/store/pos_store";
+import { patch } from "@web/core/utils/patch";
+
+patch(PosStore.prototype, {
+    // @Override
+    async _processData(loadedData) {
+        await super._processData(...arguments);
+        this.account_tax_group_by_id = loadedData['account_tax_group_by_id']
```
Comment (caburj) [inline addons/l10n_in_pos/static/src/overrides/models/pos_store.js:10]: This seems to be unused.

**Comment by hamo-odoo** [comment]
@caburj @qle-odoo Let's move forward with this PR

**Review by caburj — APPROVED** [review]
robodoo r+

**Comment by hamo-odoo** [comment]
@caburj upto 17.0

**Comment by hamo-odoo** [comment]
@fw-bot upto 17.0

**Comment by hamo-odoo** [comment]
@fw-bot up to 17.0

**Pull request closed without merging**